doc/userguide/rules/datasets.rst

```diff
@@ -96,6 +96,23 @@ hashsize <size>
 
 .. note:: 'load' and 'state' or 'save' and 'state' cannot be mixed.
 
+Example rules could look like:
+
+1. Detect unique User-Agents::
+
+    alert http any any -> any any (msg:"LOCAL HTTP new UA";http.user_agent;dataset:set,http-ua-seen, type string, state http-ua-seen.csv; sid:8000001; rev:1;)
+
+2. Detect unique TLDs::
+
+    alert dns $HOME_NET any -> any any (msg:"LOCAL DNS unique TLD"; dns.query; pcrexform:"\.([^\.]+)$"; dataset:set,dns-tld-seen, type string, state dns-tld-seen.csv; sid:8000002; rev:1;)
+
+.. image:: dataset-examples/detect-unique-tlds.png
```

Comment:
While I like using the image to help visualize, I think we're missing either a short intro text to it or a caption, to help guide visualization.